clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/view/OpenstackSecurityGroupProvider.groovy

```diff
@@ -75,67 +54,55 @@ class OpenstackSecurityGroupProvider implements SecurityGroupProvider<OpenstackS
 
   @Override
   Set<OpenstackSecurityGroup> getAll(boolean includeRules) {
-    getSecurityGroups(includeRules)
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey('*', '*', '*', '*'), includeRules)
   }
 
   @Override
   Set<OpenstackSecurityGroup> getAllByRegion(boolean includeRules, String region) {
-    getSecurityGroups(includeRules).findAll { it.region == region }
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey('*', '*', '*', region), includeRules)
   }
 
   @Override
   Set<OpenstackSecurityGroup> getAllByAccount(boolean includeRules, String account) {
-    getSecurityGroups(includeRules).findAll { it.accountName == account }
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey('*', '*', account, '*'), includeRules)
   }
 
   @Override
   Set<OpenstackSecurityGroup> getAllByAccountAndName(boolean includeRules, String account, String name) {
-    getAllByAccount(includeRules, account).findAll { it.name == name }
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(name, '*', account, '*'), includeRules)
   }
 
   @Override
-  Set<OpenstackSecurityGroup> getAllByAccountAndRegion(boolean includeRule, String account, String region) {
-    getAllByAccount(includeRule, account).findAll { it.region == region }
+  Set<OpenstackSecurityGroup> getAllByAccountAndRegion(boolean includeRules, String account, String region) {
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey('*', '*', account, region), includeRules)
   }
 
   @Override
   OpenstackSecurityGroup get(String account, String region, String name, String vpcId) {
-    getAllByAccountAndRegion(true, account, region).find { it.name == name }
+    getAllMatchingKeyPattern(Keys.getSecurityGroupKey(name, '*', account, region), true)[0]
```

Comment:
Just returns an empty collection, but I will add a test for that case to be sure.